model/project_ref.go

```diff
@@ -247,16 +247,55 @@ func (p *ProjectRef) Add(creator *user.DBUser) error {
 	return p.AddPermissions(creator)
 }
 
+func (p *ProjectRef) GetRepoId() string {
+	return fmt.Sprintf("%s_%s", p.Owner, p.Repo)
+}
+
+func (p *ProjectRef) HasEditPermission(u *user.DBUser) bool {
+	if utility.StringSliceContains(p.Admins, u.Username()) {
+		return true
+	}
+	isBranchAdmin := u.HasPermission(gimlet.PermissionOpts{
```

Comment:
I'm making this comment before reading the rest of the pr, but the permissions framework handles things more gracefully than what I think this is doing. You can define a scope called evergreen for the project admin. That scope can include evergreen_dev, evergreen_release, etc. sub-scopes for the branch admins. Having permission for evergreen will give you permission for all of its sub-scopes, you just need to write code that maintains this relation

Reply:
Okay yeah, because we're going to be adding the branch IDs to the repo scope resources, I think we don't need to check the repo resource specifically for branch permissions.